[website] Add pricing page

siriwatknp opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

**Preview**: https://deploy-preview-27598--material-ui.netlify.app/branding/pricing/

**What's left**
- [x] ~What should be the link for "Get Started"~ use same links as previous rebranding.
- [x] Info inside advanced section
- [x] https://validator.w3.org/nu/?doc=https%3A%2F%2Fdeploy-preview-27598--material-ui.netlify.app%2Fbranding%2Fpricing%2F
- [x] Lighthouse score https://developers.google.com/speed/pagespeed/insights/?url=https%3A%2F%2Fdeploy-preview-27598--material-ui.netlify.app%2Fbranding%2Fpricing%2F&tab=desktop
  
<img width="487" alt="Screen Shot 2564-08-09 at 15 25 44" src="https://user-images.githubusercontent.com/18292247/128678214-23219bdb-4049-4176-8e73-e2f4be630ed5.png">


---
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

oliviertassinari commented on docs/pages/branding/pricing-old.tsx:
```diff
@@ -0,0 +1,650 @@
+import * as React from 'react';
```
We can kill it

oliviertassinari commented on docs/src/components/pricing/EarlyBird.tsx:
```diff
@@ -0,0 +1,37 @@
+          alignItems: 'center',
+        }}
+      >
+        <Box>
```
```suggestion
        <div>
```

oliviertassinari commented on docs/src/components/pricing/PricingList.tsx:
```diff
@@ -0,0 +1,94 @@
+      {benefits &&
+        benefits.map((text) => (
+          <Box key={text} sx={{ display: 'flex', alignItems: 'center', mt: 2 }}>
+            <img src={`/static/branding/pricing/yes-${mode}.svg`} alt="" />
```
I would challenge the fact that we ever compose the image URLs. My concern is that it will make it a lot harder to remove assets, we have no guarantee when grepping that it's not used.

Also, the usage of `/static` is from a constraint of 2018 when I first introduce Next.js. Now, we could evaluate if importing them in JS is more relevant, if we do we wouldn't be able to create URLs dynamically either.

siriwatknp replied:
I will keep this in backlog for refinement, given it is the implementation detail.

oliviertassinari commented on docs/src/components/pricing/PricingTable.tsx:
```diff
@@ -0,0 +1,486 @@
+    <ColumnHead
+      {...{
+        label: '@mui/system',
+        tooltip: 'The unstyled and headless (hooks) components version of @material-ui/core.',
```
nop

oliviertassinari commented on docs/src/components/pricing/HeroPricing.tsx:
```diff
@@ -0,0 +1,34 @@
+import Container from '@material-ui/core/Container';
+import Divider from '@material-ui/core/Divider';
+import Typography from '@material-ui/core/Typography';
+import GradientText from '../typography/GradientText';
```
absolute path
```suggestion
import GradientText from 'docs/src/components/typography/GradientText';
```

x the others

oliviertassinari reviewed (commented):
A quick review:

- There are a couple of issues with the colors, e.g. @mui/unstyled is grey 800 instead of 900
- I struggle to get along with the look of the tooltip. My brain is wired for tooltip === dark background.
- Are we sure about the approach on mobile? I would personally advocate for keeping the table, make it horiontaly scrollable in the worse case scenario.

danilo-leal commented:
@oliviertassinari 

> I struggle to get along with the look of the tooltip. My brain is wired for tooltip === dark background.

Haha! Is it bothering you in any way? I mean, is it making anything hard to read or something? Check [Radix tooltip](https://www.radix-ui.com/docs/primitives/components/avatar) as well and see if it shakes you too.

> Are we sure about the approach on mobile? I would personally advocate for keeping the table, make it horizontally scrollable in the worst-case scenario.

Jun hasn't implemented the latest iteration of the pricing table yet. [I'm pushing this one forward.](https://rebranding-demo.webflow.io/pricing-table-demo) Check the responsiveness and let me know what you think. It has a lot of minor improvements over the currently implemented one.

oliviertassinari commented:
> Check Radix tooltip as well and see if it shakes you too.

@danilo-leal There are two, I personally enjoy the UX in both

<img width="255" alt="Capture d’écran 2021-08-06 à 01 20 12" src="https://user-images.githubusercontent.com/3165635/128433541-c8059b27-077a-4d2e-8409-b53c64b62814.png">

<img width="330" alt="Capture d’écran 2021-08-06 à 01 20 17" src="https://user-images.githubusercontent.com/3165635/128433539-18f843f4-e27f-433b-909c-2ea68441873c.png">

siriwatknp commented:
> > Check Radix tooltip as well and see if it shakes you too.
> 
> @danilo-leal I can find two different ones. I personally enjoy the UX in both:
> 

What's about we add shadow to the current tooltip? @danilo-leal

danilo-leal commented:
> > > Check Radix tooltip as well and see if it shakes you too.
> > 
> > 
> > @danilo-leal I can find two different ones. I personally enjoy the UX in both:
> 
> What's about we add shadow to the current tooltip? @danilo-leal

I think it works! No problem with that. We can add maybe the mid-level shadow (see styles on Figma).

oliviertassinari commented:
I have spent 1 hour on this PR, you can review the commits I have pushed. I have tried to polish stuff up. It looks great.

oliviertassinari commented:
1. The page should have an h1, maybe this one?

<img width="546" alt="Capture d’écran 2021-08-10 à 00 56 54" src="https://user-images.githubusercontent.com/3165635/128784514-e68bb1dc-cb47-442b-a83f-1e758cc36da6.png">

2. I'm not sure if h3 is correct at this level

<img width="432" alt="Capture d’écran 2021-08-10 à 00 56 47" src="https://user-images.githubusercontent.com/3165635/128784535-0595a1b4-8f61-42de-b054-0f7386f023e9.png">

3. "Material-UI X" should become "MUI X" in the copywriting?
4. Outdated wording?

<img width="494" alt="Capture d’écran 2021-08-10 à 01 00 59" src="https://user-images.githubusercontent.com/3165635/128784825-a082e96b-8eb9-42a4-a04f-d1f0e9b50251.png">

5. The different prices not on the same horizontal line feels weird

<img width="897" alt="lines" src="https://user-images.githubusercontent.com/3165635/128785100-e7b20416-d243-4a3f-a32c-27ff441f0138.png">

siriwatknp commented:
@oliviertassinari
> 7. I think that there is a "weight" issue with the yes/no checkmarks in the pricing table. This is the current version:

I think check should be a little bit bigger?

oliviertassinari commented:
@siriwatknp it seems common (in other popular products) to leave the no cells blank, with no no-icon. I didn't experiment with the size, maybe it helps.

siriwatknp commented:
@danilo-leal any thought? otherwise, we can merge this as a first iteration and do the improvement again (there are some code that I need for the homepage improvement).

danilo-leal commented:
> @danilo-leal any thought? otherwise, we can merge this as a first iteration and do the improvement again (there are some code that I need for the homepage improvement).

I think what we can do is darken a little bit the check blue and make the X a little lighter (had received other feedback saying that the first iteration of the X icon seemed like it was disabled. Which, to me, wasn't so much of a big deal since it is, after all, not interactive. But...)

danilo-leal commented:
It's looking amazing! Very nice work, Jun! 😍
I have just two feedbacks:

1. About the Data Grid row: are we going to add the icons there? And it's a bit weird to me that only this row is missing the recommended column background.

<img width="1170" alt="Screen Shot 2021-08-10 at 09 57 12" src="https://user-images.githubusercontent.com/67129314/128871312-cae2a41c-c017-46ac-86f0-f404cbcf208b.png">

2. I think we should standardize the inline link styles, we use two different ones on the Data Grid features and on the FAQ box there. I personally would prefer we use the first one.

<img width="400" alt="Screen Shot 2021-08-10 at 10 00 17" src="https://user-images.githubusercontent.com/67129314/128871295-330f1151-ef6a-4325-b9d8-db598c6f7fb1.png">

siriwatknp commented:
> 1. About the Data Grid row: are we going to add the icons there? And it's a bit weird to me that only this row is missing the recommended column background.

Added highlighted background but I think leaving the cell blank makes it different from usual cell which is better because it has interaction.

> 2. I think we should standardize the inline link styles, we have two different ones going in this page: one on the Data Grid features and another on the FAQ box there. I personally would prefer we use the first.

I changed the weight of the links to `500`, I think it looks better (not too bold but still obvious among normal text).
